server/src/updaters/thread-permission-updaters.js

```diff
@@ -282,16 +308,24 @@ async function updateDescendantPermissions(
           permissions,
         );
         if (permissions) {
-          changeset.push({
+          membershipRows.push({
             operation: 'update',
             userID,
             threadID,
             permissions,
             permissionsForChildren,
             role,
           });
+
+          for (const [childUserID] of userInfos) {
+            if (childUserID !== userID) {
+              const [user1, user2] = sortIDs(childUserID, userID);
+              const status = undirectedStatus.KNOW_OF;
+              relationshipRows.push({ user1, user2, status });
+            }
+          }
```

Comment:
Sorry for not noticing this earlier, but I think we can safely gate this behind an `if (!oldPermissions) {` condition